docs/resources/service_object.md

```diff
@@ -116,18 +116,17 @@ resource "valtix_service_object" "forward-https" {
 
 For a complete set of arguments, see [Forwarding Arguments](#forwarding-arguments)
 
-## Argument Reference
-
-### ReverseProxy Arguments
-* `name` - (Required) Name of the Service Object
-* `description` - (Optional) Description of the Service Object
+## ReverseProxy Arguments
+* `name` - (Required) Name of the service object
+* `description` - (Optional) Description of the service object
 * `service_type` - (Required) `ReverseProxy`
-* `protocol` - (Optional) `TCP` or `UDP`. If not specified, the default value is `TCP`.
-* `transport_mode` - (Required) `HTTP`, `HTTPS`, `WEBSOCKET_S`, `TCP`, `TLS`. The protocol used by the Gateway to communicate with the backend application defined using a Reverse Proxy Target Address Object (`backend_address_group`).
-* `port` - (Required) This argument can be specified multiple times if the service runs on multiple ports. Structure is [documented below](#reverseproxy-port)
+* `protocol` - (Optional) `TCP` or `UDP`. `TCP` is default. This is the listener protocol.
+* `transport_mode` - (Required) `HTTPS`, `TLS`, `WEBSOCKET_S` if tls_profile is set else `HTTP`, `TCP`, `WEBSOCKET`. The protocol used by the Gateway to communicate with the backend application defined using a Reverse Proxy Target Address Object (`backend_address_group`)
```

Comment:
If tls_profile is specified, then the valid values are https tls and websocket_s. If tls_profile is not specified then the valiud values are http tcp and websocket